pkg/controller/rollout/canary.go

```diff
@@ -45,13 +45,21 @@ func (r *rolloutContext) runCanary() error {
 	// update canary status
 	batch, err := r.batchControl.FetchBatchRelease()
 	if err != nil {
+		canaryStatus.Message = "BatchRelease not found"
 		canaryStatus.CanaryReplicas = r.workload.CanaryReplicas
 		canaryStatus.CanaryReadyReplicas = r.workload.CanaryReadyReplicas
 	} else {
+		canaryStatus.Message = fmt.Sprintf("BatchRelease at state %s, id %s, step %d",
+			batch.Status.CanaryStatus.CurrentBatchState, batch.Status.ObservedRolloutID, batch.Status.CanaryStatus.CurrentBatch+1)
 		canaryStatus.CanaryReplicas = batch.Status.CanaryStatus.UpdatedReplicas
 		canaryStatus.CanaryReadyReplicas = batch.Status.CanaryStatus.UpdatedReadyReplicas
 	}
 
+	// sync rollout-id to batchRelease if we need
+	if err := r.batchControl.SyncRolloutID(getRolloutID(r.workload, r.rollout)); err != nil {
```

Comment:
What is the canaryStatus.RolloutId not sync? I think it should sync from canaryStatus.RolloutId to br.

Reply:
done